### PR TITLE
[REVIEW] Cleanup 0.14

### LINKED ISSUE #641: Clean dead code

### PR DESCRIPTION
**Pull request by jeanp413**
Resolved #641

### CODE REVIEW

**Comment by kharoc** [comment]
1 RAL

========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1261/1274
FAILED: 13/1274
CRASH: 0/1274
TOTAL: 1274
MAX DELTA: 212.0
*******************************************************************************
Aggregations without group by:   Start Mem: 437.875   End Mem: 559.875   Diff: 122.0
Coalesce:   Start Mem: 559.875   End Mem: 509.875   Diff: -50.0
Column Basis:   Start Mem: 509.875   End Mem: 509.875   Diff: 0.0
Common Table Expressions:   Start Mem: 509.875   End Mem: 607.875   Diff: 98.0
Count without group by:   Start Mem: 607.875   End Mem: 617.875   Diff: 10.0
Date:   Start Mem: 617.875   End Mem: 661.875   Diff: 44.0
Timestamp:   Start Mem: 661.875   End Mem: 683.875   Diff: 22.0
Full outer join:   Start Mem: 683.875   End Mem: 669.875   Diff: -14.0
Group by:   Start Mem: 669.875   End Mem: 669.875   Diff: 0.0
Group by without aggregations:   Start Mem: 669.875   End Mem: 793.875   Diff: 124.0
Inner join:   Start Mem: 793.875   End Mem: 825.875   Diff: 32.0
Left outer join:   Start Mem: 825.875   End Mem: 815.875   Diff: -10.0
Non-EquiJoin Queries:   Start Mem: 815.875   End Mem: 761.875   Diff: -54.0
Nested Queries:   Start Mem: 761.875   End Mem: 765.875   Diff: 4.0
Order by:   Start Mem: 765.875   End Mem: 785.875   Diff: 20.0
Predicates With Nulls:   Start Mem: 785.875   End Mem: 833.875   Diff: 48.0
Simple String:   Start Mem: 833.875   End Mem: 863.875   Diff: 30.0
Unary ops:   Start Mem: 1073.875   End Mem: 1021.875   Diff: -52.0
Unify Tables:   Start Mem: 1021.875   End Mem: 1039.875   Diff: 18.0
Union:   Start Mem: 1039.875   End Mem: 1077.875   Diff: 38.0
Limit:   Start Mem: 1077.875   End Mem: 1101.875   Diff: 24.0
Where clause:   Start Mem: 1101.875   End Mem: 1021.875   Diff: -80.0
Bindable Alias:   Start Mem: 1021.875   End Mem: 1117.875   Diff: 96.0
Boolean:   Start Mem: 1117.875   End Mem: 1175.875   Diff: 58.0
Case:   Start Mem: 1175.875   End Mem: 1259.875   Diff: 84.0
Cast:   Start Mem: 1259.875   End Mem: 1345.875   Diff: 86.0
Concat:   Start Mem: 1345.875   End Mem: 1361.875   Diff: 16.0
Literal:   Start Mem: 1361.875   End Mem: 1419.875   Diff: 58.0
Dir:   Start Mem: 1419.875   End Mem: 1547.875   Diff: 128.0
File System Local:   Start Mem: 1547.875   End Mem: 1541.875   Diff: -6.0
File System S3:   Start Mem: 1541.875   End Mem: 1683.875   Diff: 142.0
File System Google Storage :   Start Mem: 1683.875   End Mem: 1807.875   Diff: 124.0
Like:   Start Mem: 1807.875   End Mem: 1873.875   Diff: 66.0
Substring:   Start Mem: 2249.875   End Mem: 2279.875   Diff: 30.0
Wild Card:   Start Mem: 2279.875   End Mem: 2421.875   Diff: 142.0
TPCH Queries:   Start Mem: 2421.875   End Mem: 2633.875   Diff: 212.0
Round:   Start Mem: 2633.875   End Mem: 2727.875   Diff: 94.0

2 Worker's

========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1500/1513
FAILED: 13/1513
CRASH: 0/1513
TOTAL: 1513
MAX DELTA: 483.0
*******************************************************************************
Aggregations without group by:   Start Mem: 864.875   End Mem: 1347.875   Diff: 483.0
Coalesce:   Start Mem: 1347.875   End Mem: 1525.875   Diff: 178.0
Common Table Expressions:   Start Mem: 1525.875   End Mem: 1531.875   Diff: 6.0
Count without group by:   Start Mem: 1531.875   End Mem: 1529.875   Diff: -2.0
Date:   Start Mem: 1529.875   End Mem: 1527.875   Diff: -2.0
Timestamp:   Start Mem: 1527.875   End Mem: 1535.875   Diff: 8.0
Full outer join:   Start Mem: 1535.875   End Mem: 1625.875   Diff: 90.0
Group by:   Start Mem: 1625.875   End Mem: 1617.875   Diff: -8.0
Group by without aggregations:   Start Mem: 1617.875   End Mem: 1693.875   Diff: 76.0
Inner join:   Start Mem: 1693.875   End Mem: 1699.875   Diff: 6.0
Left outer join:   Start Mem: 1699.875   End Mem: 1707.875   Diff: 8.0
Non-EquiJoin Queries:   Start Mem: 1707.875   End Mem: 1743.875   Diff: 36.0
Nested Queries:   Start Mem: 1743.875   End Mem: 1885.875   Diff: 142.0
Order by:   Start Mem: 1885.875   End Mem: 1865.875   Diff: -20.0
Predicates With Nulls:   Start Mem: 1865.875   End Mem: 1965.875   Diff: 100.0
Simple String:   Start Mem: 1965.875   End Mem: 1939.875   Diff: -26.0
Unary ops:   Start Mem: 2259.875   End Mem: 2299.875   Diff: 40.0
Unify Tables:   Start Mem: 2299.875   End Mem: 2439.875   Diff: 140.0
Union:   Start Mem: 2439.875   End Mem: 2425.875   Diff: -14.0
Limit:   Start Mem: 2425.875   End Mem: 2319.875   Diff: -106.0
Where clause:   Start Mem: 2319.875   End Mem: 2369.875   Diff: 50.0
Bindable Alias:   Start Mem: 2369.875   End Mem: 2527.875   Diff: 158.0
Boolean:   Start Mem: 2527.875   End Mem: 2565.875   Diff: 38.0
Case:   Start Mem: 2565.875   End Mem: 2717.875   Diff: 152.0
Cast:   Start Mem: 2717.875   End Mem: 2715.875   Diff: -2.0
Concat:   Start Mem: 2715.875   End Mem: 2993.875   Diff: 278.0
Literal:   Start Mem: 2993.875   End Mem: 2953.875   Diff: -40.0
Dir:   Start Mem: 2953.875   End Mem: 3347.875   Diff: 394.0
File System Local:   Start Mem: 3347.875   End Mem: 3405.875   Diff: 58.0
File System S3:   Start Mem: 3405.875   End Mem: 3603.875   Diff: 198.0
File System Google Storage :   Start Mem: 3603.875   End Mem: 3779.875   Diff: 176.0
Like:   Start Mem: 3779.875   End Mem: 3969.875   Diff: 190.0
Substring:   Start Mem: 4879.875   End Mem: 5047.875   Diff: 168.0
Wild Card:   Start Mem: 5047.875   End Mem: 5387.875   Diff: 340.0
TPCH Queries:   Start Mem: 5387.875   End Mem: 5707.875   Diff: 320.0
Round:   Start Mem: 5707.875   End Mem: 5825.875   Diff: 118.0

File: engine/src/CalciteExpressionParsing.cpp [hunk 1]
```diff
@@ -258,42 +251,19 @@ cudf::type_id get_output_type_expression(const cudf::table_view & table, std::st
...
-						"zero elements");
+					RAL_FAIL("In function get_output_type_expression, the operator cannot be processed on less than one or zero elements");
 
 				cudf::type_id left_operand = operands.top();
 				operands.pop();
 				cudf::type_id right_operand = operands.top();
 				operands.pop();
 
-				// TODO percy cudf0.12 was invalid here, should we consider empty?
-				if(left_operand == cudf::type_id::EMPTY) {
```
Comment (williamBlazing) [inline engine/src/CalciteExpressionParsing.cpp:271]: porque se borro este codigo? Asumo que nunca deberia de suceder, pero si sucede por alguna razon no esperada, seria util obtener un error deterministico que nos ayude a saber rapidamente que paso

File: engine/src/CalciteInterpreter.cpp [hunk 2]
```diff
@@ -431,7 +148,6 @@ std::unique_ptr<ral::frame::BlazingTable> execute_plan(std::vector<ral::io::data
 									"substep"_a=queryContext.getQuerySubstep(),
 									"info"_a="In execute_plan. What: {}"_format(e.what()),
 									"duration"_a="");
-		logger->flush();
```
Comment (williamBlazing) [inline engine/src/CalciteInterpreter.cpp:434]: porque borraste ese flush? la idea es que cuando hay un error, se le haga flush, para asegurar que se escribe al archivo antes de que crashee el proceso

File: engine/src/CalciteExpressionParsing.cpp [hunk 3]
```diff
@@ -258,42 +251,19 @@ cudf::type_id get_output_type_expression(const cudf::table_view & table, std::st
...
-						"In function get_output_type_expression, the operator cannot be processed on less than one or "
-						"zero elements");
+					RAL_FAIL("In function get_output_type_expression, the operator cannot be processed on less than one or zero elements");
 
 				cudf::type_id left_operand = operands.top();
 				operands.pop();
 				cudf::type_id right_operand = operands.top();
 				operands.pop();
 
-				// TODO percy cudf0.12 was invalid here, should we consider empty?
```
Comment (williamBlazing) [inline engine/src/CalciteExpressionParsing.cpp:270]: Porque se borro esto? Si los types EMPTY nunca deberian de suceder seria bueno, por lo menos que obtengamos un error predecible cuando eso suceda

Comment (williamBlazing) [reply engine/src/CalciteExpressionParsing.cpp:270]: Ahh creo que entiendo, esta verificacion la estas haciendo dentro de otra funcion

File: engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h [hunk 4]
```diff
@@ -55,12 +55,11 @@ class SortAndSampleSingleNodeKernel : public kernel {
 											"substep"_a=context->getQuerySubstep(),
 											"info"_a="In SortAndSampleSingleNode kernel batch {} for {}. What: {}"_format(batch_count, expression, e.what()),
 											"duration"_a="");
-				logger->flush();
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h:58]: porque estas quitando este flush?

File: engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h [hunk 5]
```diff
@@ -123,7 +122,6 @@ class PartitionSingleNodeKernel : public kernel {
 											"substep"_a=context->getQuerySubstep(),
 											"info"_a="In PartitionSingleNode kernel batch {} for {}. What: {}"_format(batch_count, expression, e.what()),
 											"duration"_a="");
-				logger->flush();
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h:126]: porque estas quitando el flush?

File: engine/src/communication/messages/ComponentMessages.h [hunk 6]
```diff
@@ -8,9 +8,8 @@
 namespace ral {
 namespace communication {
 namespace messages {
-namespace experimental {
-
 
+using blazingdb::transport::experimental::Node;
```
Comment (williamBlazing) [inline engine/src/communication/messages/ComponentMessages.h:12]: Ya que has limpiado los experimental namespaces muy bien, podrias limpiarlos de estas clases tambien

File: engine/src/CalciteExpressionParsing.cpp (hunk 3 above)
Comment (jeanp413) [reply engine/src/CalciteExpressionParsing.cpp:270]: yeah that check now is done in `get_output_type`

File: engine/src/CalciteExpressionParsing.cpp (hunk 1 above)
Comment (jeanp413) [reply engine/src/CalciteExpressionParsing.cpp:271]: that check now is done in `get_output_type`

File: engine/src/CalciteInterpreter.cpp (hunk 2 above)
Comment (jeanp413) [reply engine/src/CalciteInterpreter.cpp:434]: I'm calling `spdlog::flush_on(spdlog::level::err);` in `initialize.cpp` which does exactly the same

File: engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h (hunk 4 above)
Comment (jeanp413) [reply engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h:58]: I'm calling `spdlog::flush_on(spdlog::level::err);` in `initialize.cpp` which does exactly the same

File: engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h (hunk 5 above)
Comment (jeanp413) [reply engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h:126]: same as above

**Comment by kharoc** [comment]
1 RAL

========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1261/1274
FAILED: 13/1274
CRASH: 0/1274
TOTAL: 1274
MAX DELTA: 48.0
*******************************************************************************
Aggregations without group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Coalesce:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Column Basis:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Common Table Expressions:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Count without group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Date:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Timestamp:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Full outer join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Group by without aggregations:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Inner join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Left outer join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Non-EquiJoin Queries:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Nested Queries:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Order by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Predicates With Nulls:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Simple String:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Unary ops:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Unify Tables:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Union:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Limit:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Where clause:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Bindable Alias:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Boolean:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Case:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Cast:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Concat:   Start Mem: 8117.875   End Mem: 8165.875   Diff: 48.0
Literal:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Dir:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System Local:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System S3:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System Google Storage :   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Like:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Substring:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Wild Card:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
TPCH Queries:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Round:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0

2 Worker's

========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1500/1513
FAILED: 13/1513
CRASH: 0/1513
TOTAL: 1513
MAX DELTA: 457.0
*******************************************************************************
Aggregations without group by:   Start Mem: 864.875   End Mem: 1321.875   Diff: 457.0
Coalesce:   Start Mem: 1321.875   End Mem: 1409.875   Diff: 88.0
Common Table Expressions:   Start Mem: 1409.875   End Mem: 1411.875   Diff: 2.0
Count without group by:   Start Mem: 1411.875   End Mem: 1431.875   Diff: 20.0
Date:   Start Mem: 1431.875   End Mem: 1517.875   Diff: 86.0
Timestamp:   Start Mem: 1517.875   End Mem: 1531.875   Diff: 14.0
Full outer join:   Start Mem: 1531.875   End Mem: 1545.875   Diff: 14.0
Group by:   Start Mem: 1545.875   End Mem: 1583.875   Diff: 38.0
Group by without aggregations:   Start Mem: 1583.875   End Mem: 1595.875   Diff: 12.0
Inner join:   Start Mem: 1595.875   End Mem: 1653.875   Diff: 58.0
Left outer join:   Start Mem: 1653.875   End Mem: 1675.875   Diff: 22.0
Non-EquiJoin Queries:   Start Mem: 1675.875   End Mem: 1783.875   Diff: 108.0
Nested Queries:   Start Mem: 1783.875   End Mem: 1813.875   Diff: 30.0
Order by:   Start Mem: 1813.875   End Mem: 1845.875   Diff: 32.0
Predicates With Nulls:   Start Mem: 1845.875   End Mem: 1895.875   Diff: 50.0
Simple String:   Start Mem: 1895.875   End Mem: 1961.875   Diff: 66.0
Unary ops:   Start Mem: 2281.875   End Mem: 2335.875   Diff: 54.0
Unify Tables:   Start Mem: 2335.875   End Mem: 2415.875   Diff: 80.0
Union:   Start Mem: 2415.875   End Mem: 2397.875   Diff: -18.0
Limit:   Start Mem: 2397.875   End Mem: 2273.875   Diff: -124.0
Where clause:   Start Mem: 2273.875   End Mem: 2363.875   Diff: 90.0
Bindable Alias:   Start Mem: 2363.875   End Mem: 2505.875   Diff: 142.0
Boolean:   Start Mem: 2505.875   End Mem: 2545.875   Diff: 40.0
Case:   Start Mem: 2545.875   End Mem: 2675.875   Diff: 130.0
Cast:   Start Mem: 2675.875   End Mem: 2761.875   Diff: 86.0
Concat:   Start Mem: 2761.875   End Mem: 2967.875   Diff: 206.0
Literal:   Start Mem: 2967.875   End Mem: 2979.875   Diff: 12.0
Dir:   Start Mem: 2979.875   End Mem: 3349.875   Diff: 370.0
File System Local:   Start Mem: 3349.875   End Mem: 3377.875   Diff: 28.0
File System S3:   Start Mem: 3377.875   End Mem: 3553.875   Diff: 176.0
File System Google Storage :   Start Mem: 3553.875   End Mem: 3731.875   Diff: 178.0
Like:   Start Mem: 3731.875   End Mem: 3997.875   Diff: 266.0
Substring:   Start Mem: 4917.875   End Mem: 5055.875   Diff: 138.0
Wild Card:   Start Mem: 5055.875   End Mem: 5475.875   Diff: 420.0
TPCH Queries:   Start Mem: 5475.875   End Mem: 5837.875   Diff: 362.0
Round:   Start Mem: 5837.875   End Mem: 5937.875   Diff: 100.0

**Pull request merged**